drafts/25-xxx-specifications.txt

```diff
@@ -555,16 +555,59 @@ nd20. The result of evaluating a processor-dependent directive is processor-depe
 4 Macro recognition and expansion
 ---------------------------------
 
+This section is currently incomplete.  A self-contained specification
+for macro recognition and expansion will appear in a forthcoming
+paper.
+
+4.1 Comparison to Macro recognition and expansion in CPP
+--------------------------------------------------------
+
+We intend for macro recognition and expansion to operate almost
+exactly as specified in C 2023 section 6.10.5, with the following
+minor exceptions:
+
+me10. When determining argument boundaries in the invocation of a
+      function-like macro, FPP ignores commas surrounded by matching
+      sets of `[  ]` or `(/  /)` brackets, in addition to matching
+      sets of `(  )` parentheses (as specified in C 2023 section
+      6.10.5-11).
+
+me30. As specified by C 2023 6.10.5-10: "Within the sequence of
+      preprocessing tokens making up an invocation of a function-like
+      macro, new-line is considered a normal white-space character."
```

Comment:
I don't think we should have rule m30. It makes switching between function-like macros and function calls incompatible. According to this rule,
```fortran
#define atan(x,y) ATAN(x,y)
      real :: g = atan(x,
                          y)
```
and
```fortran
      real :: g = atan(x,&
                           y)
```
asymmetric. The `&` is optional in the FLM case, but required in the function case.

I think they should be required in both cases.